src/WebJobs.Script/DependencyInjection/ScriptStartupTypeLocator.cs

```diff
@@ -52,6 +52,7 @@ public class ScriptStartupTypeLocator : IWebJobsStartupTypeLocator
             _metricsLogger = metricsLogger;
             _startupTypes = new Lazy<IEnumerable<Type>>(() => GetExtensionsStartupTypesAsync().ConfigureAwait(false).GetAwaiter().GetResult());
             _languageWorkerOptions = languageWorkerOptions;
+            _extensionRequirements = DependencyHelper.GetExtensionRequirements(SystemEnvironment.Instance);
```

Comment:
Should we consider injecting `IEnvironment`  via `ScriptStartupTypeLocator` constructor and use that instead of using the concrete implementation directly here?

Reply:
Can't really do that since `IEnvironment` it is not available yet. ScriptStartupTypeLocator in manually instantiated during the app configuration stage.